### PR TITLE
add exercise zebra-puzzle

### PR DESCRIPTION
**Pull request by siebenschlaefer**
Feel free to criticize anything, big or small.

### CODE REVIEW

File: exercises/practice/zebra-puzzle/zebra_puzzle.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef ZEBRA_PUZZLE_H
+#define ZEBRA_PUZZLE_H
+
+/// Determine the nationality of the resident who drinks water.
+const char *drinks_water(void);
+
+/// Determine the nationality of the resident who owns the zebra.
+const char *owns_zebra(void);
+
+#endif // ZEBRA_PUZZLE_H
```
Comment (ryanplusplus) [inline exercises/practice/zebra-puzzle/zebra_puzzle.h:10]:
```suggestion
#endif
```

Removed for consistency with other exercises

File: config.json [hunk 2]
```diff
@@ -1050,6 +1050,14 @@
...
       },
       {
+        "slug": "zebra-puzzle",
+        "name": "Zebra Puzzle",
+        "uuid": "2cae2796-5c05-4159-b3e7-9f5653f0bc9b",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 6
+      },
+      }
```
Comment (ryanplusplus) [inline config.json:1060]:
```suggestion
      {
```

My bad, I messed this up fixing the conflict

**Review by ryanplusplus — APPROVED** [review]
This is really slick! This is my first time seeing a solution to this exercise.

File: exercises/practice/zebra-puzzle/.meta/example.c [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+   switch (owns_zebra) {
+      case ENGLISH:
+         *out_owns_zebra = "English";
+         break;
+      case JAPANESE:
+         *out_owns_zebra = "Japanese";
+         break;
+      case NORWEGIAN:
+         *out_owns_zebra = "Norwegian";
+         break;
```
Comment (wolf99) [inline exercises/practice/zebra-puzzle/.meta/example.c:179]: Why solve the same problem twice?
Is it possible to solve it once and then just query the results (possibly in some generic way) ?

**Review by wolf99 — COMMENTED** [review]
I feel like there may be some way to implement a large part of this with x macros. But I can't see it well enough on mobile to know for sure

File: exercises/practice/zebra-puzzle/.meta/example.c (hunk 3 above)
Comment (ryanplusplus) [reply exercises/practice/zebra-puzzle/.meta/example.c:179]: Do you mean to take an object oriented approach to the exercise where solving and querying the solution are separate steps?

Comment (siebenschlaefer) [reply exercises/practice/zebra-puzzle/.meta/example.c:179]: I agree, and don't like it either.  
I was under the impression that these two separate functions would be a what the author of the [problem-specification](https://github.com/exercism/problem-specifications/blob/main/exercises/zebra-puzzle/canonical-data.json) intended, with the two "properties" `drinksWater` and `ownsZebra`.  
I looked at the [C# version](https://github.com/exercism/csharp/blob/main/exercises/practice/zebra-puzzle/ZebraPuzzle.cs) where those are two `static` methods and the [Python version](https://github.com/exercism/python/blob/main/exercises/practice/zebra-puzzle/zebra_puzzle.py) where those are two free functions.

But now that you asked I dug a little further and found the [Go version](https://github.com/exercism/go/blob/main/exercises/practice/zebra-puzzle/zebra_puzzle.go) that has a single function `SolvePuzzle()` which I like more.

I could change the code to something like
```c
struct solution
{
   const char *drinks_water;
   const char *owns_zebra;
};

struct solution solve_puzzle(void);
```

**Pull request merged**